### PR TITLE
docs: Refactor documentation for GameWidget

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Use the `dartdoc` directive for the `GameWidget` class, allowing us to keep the documentation in the code and on the site in sync.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [-] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/game/game_widget/game_widget.dart [hunk 1]
```diff
@@ -8,122 +8,38 @@ import 'package:flame/src/game/game_widget/gesture_detector_builder.dart';
...
+/// }
+/// ```
+///
+/// At the same time, `GameWidget` is a regular Flutter widget, and can be
+/// inserted arbitrarily deep into the widget tree, including the possibility
+/// of having multiple `GameWidget`s within a single app.
+///
+/// The layout behavior of this widget is that it will expand to fill all
+/// available space. Thus, when used as a root widget it will make the app
+/// full-screen. Inside any other layout widget it will take as much space as
```
Comment (erickzanardo) [inline packages/flame/lib/src/game/game_widget/game_widget.dart:31]: One thing that I think we should also mention here, that `GameWidget` doesn't clip automatically. Meaning that is the user render something outside the boundaries of the parent of the game widget, the game content will "overflow", that if the user wants to clip, they should wrap the game widget into a ClipRect widget.

**Review by erickzanardo — COMMENTED** [review]
Awesome docs improvement, left one small comment

File: packages/flame/lib/src/game/game_widget/game_widget.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame/lib/src/game/game_widget/game_widget.dart:31]: Good point. Will need to add clipping to Flame Studio as well.

Comment (st-pasha) [reply packages/flame/lib/src/game/game_widget/game_widget.dart:31]: Done.

**Review by spydon — APPROVED** [review]

**Pull request merged**